src/tests/class_tests/openms/source/PeakIntegrator_test.cpp

```diff
@@ -848,6 +920,459 @@ START_SECTION(PeakShapeMetrics calculatePeakShapeMetrics(
 }
 END_SECTION
 
+START_SECTION(void fitEMGPeakModel(
+  const MSChromatogram& input_peak,
+  MSChromatogram& output_peak
+) const)
+{
+  MSChromatogram out_min, out_sec;
+  const MSChromatogram::FloatDataArray * fda_emg;
+
+  PeakIntegrator pi;
+  Param params = pi.getParameters();
+  pi.setParameters(params);
+
+  pi.fitEMGPeakModel(chromatogram, out_min);
+  TEST_EQUAL(out_min.size(), 107)
+
+  fda_emg = &out_min.getFloatDataArrays()[0];
+  TEST_EQUAL(fda_emg->getName(), "emg_parameters")
+  TEST_REAL_SIMILAR((*fda_emg)[0], 1317410)
+  TEST_REAL_SIMILAR((*fda_emg)[1], 2.68121)
+  TEST_REAL_SIMILAR((*fda_emg)[2], 0.0212625)
+  TEST_REAL_SIMILAR((*fda_emg)[3], 0.0235329)
+  // geogebra_print_execute((*fda_emg)[0], (*fda_emg)[1], (*fda_emg)[2], (*fda_emg)[3]);
+
+  pi.fitEMGPeakModel(saturated_chrom_min, out_min);
+  pi.fitEMGPeakModel(saturated_chrom_sec, out_sec);
+  TEST_EQUAL(out_min.size(), 87)
+  TEST_EQUAL(out_min.size(), out_sec.size())
+  TOLERANCE_RELATIVE(1.0 + 1e-2)
+  for (Size i = 0; i < out_min.size(); i += 9)
+  {
+    TEST_REAL_SIMILAR(out_min[i].getIntensity(), out_sec[i].getIntensity())
+  }
+  TOLERANCE_RELATIVE(1.0 + 1e-5)
+
+  fda_emg = &out_min.getFloatDataArrays()[0];
+  TEST_EQUAL(fda_emg->getName(), "emg_parameters")
+  TEST_REAL_SIMILAR((*fda_emg)[0], 7735860)
+  TEST_REAL_SIMILAR((*fda_emg)[1], 2.66296)
+  TEST_REAL_SIMILAR((*fda_emg)[2], 0.0394313)
+  TEST_REAL_SIMILAR((*fda_emg)[3], 0.0394313)
+  // geogebra_print_execute((*fda_emg)[0], (*fda_emg)[1], (*fda_emg)[2], (*fda_emg)[3]);
+
+  fda_emg = &out_sec.getFloatDataArrays()[0];
+  TEST_EQUAL(fda_emg->getName(), "emg_parameters")
+  TEST_REAL_SIMILAR((*fda_emg)[0], 7736020)
+  TEST_REAL_SIMILAR((*fda_emg)[1], 159.778)
+  TEST_REAL_SIMILAR((*fda_emg)[2], 2.36584)
+  TEST_REAL_SIMILAR((*fda_emg)[3], 2.36584)
+
+  pi.fitEMGPeakModel(saturated_cutoff_chrom_min, out_min);
+  pi.fitEMGPeakModel(saturated_cutoff_chrom_sec, out_sec);
+  TEST_EQUAL(out_min.size(), 71)
+  TEST_EQUAL(out_min.size(), out_sec.size())
+  TOLERANCE_RELATIVE(1.0 + 1e-2)
+  for (Size i = 0; i < out_min.size(); i += 9)
+  {
+    TEST_REAL_SIMILAR(out_min[i].getIntensity(), out_sec[i].getIntensity())
+  }
+  TOLERANCE_RELATIVE(1.0 + 1e-5)
+
+  fda_emg = &out_min.getFloatDataArrays()[0];
+  TEST_EQUAL(fda_emg->getName(), "emg_parameters")
+  TEST_REAL_SIMILAR((*fda_emg)[0], 15515900)
```

Comment:
Are the order of the parameters documented somewhere?  You mentioned that you store them in one of the .h docstrings, but I did not see mention of which index corresponds to which parameters.

Reply:
Added some doc in `fitEMGPeakModel()` docstrings. How does it look?